server/src/main/java/io/spine/server/aggregate/AggregateRepository.java

```diff
@@ -428,30 +428,60 @@ A loadOrCreate(I id) {
     /**
      * Loads an aggregate by the passed ID.
      *
-     * <p>To read an {@link AggregateStateRecord} from an {@link AggregateStorage},
-     * a {@linkplain #getSnapshotTrigger() snapshot trigger} is used as a
-     * {@linkplain AggregateReadRequest#getBatchSize() batch size}.
+     * <p>This method defines the basic flow of an {@code Aggregate} loading. First,
+     * the {@linkplain AggregateStateRecord Aggregate state} is {@linkplain #loadOrCreate loaded}
+     * from the storage. Then the {@code Aggregate} is {@linkplain #reify reified} from it's state.
      *
      * @param id the ID of the aggregate
-     * @return the loaded instance or {@code Optional.absent()} if there is no record with the ID
+     * @return the loaded instance or {@code Optional.absent()} if there is no {@code Aggregate}
+     *         with the ID
      */
     private Optional<A> load(I id) {
-        final AggregateReadRequest<I> request = new AggregateReadRequest<>(id, snapshotTrigger);
-        final Optional<AggregateStateRecord> eventsFromStorage = aggregateStorage().read(request);
-
+        final Optional<AggregateStateRecord> eventsFromStorage = loadState(id);
         if (eventsFromStorage.isPresent()) {
-            final A result = create(id);
-            final AggregateStateRecord aggregateStateRecord = eventsFromStorage.get();
-            final AggregateTransaction tx = AggregateTransaction.start(result);
-            result.play(aggregateStateRecord);
-            tx.commit();
+            final A result = reify(id, eventsFromStorage.get());
             return Optional.of(result);
         }
-
         return Optional.absent();
     }
 
     /**
+     * Loads the state of the {@code Aggregate} with the given ID.
+     *
+     * <p>To read an {@link AggregateStateRecord} from an {@link AggregateStorage},
+     * a {@linkplain #getSnapshotTrigger() snapshot trigger} is used as a
+     * {@linkplain AggregateReadRequest#getBatchSize() batch size}.
+     *
+     * @param id the ID of the {@code Aggregate} to load
+     * @return the {@link AggregateStateRecord} for the {@code Aggregate} or
+     *         {@code Optional.absent()} if there is no record with the ID
+     */
+    protected Optional<AggregateStateRecord> loadState(I id) {
```

Comment:
Ideally this should be called differently. There is too much `load` in this PR. We are trying to split the `load(id)` into two methods. Right now it is already split, but still a mess.